service/src/main/java/bio/terra/workspace/service/resource/controlled/cloud/aws/sagemakerNotebook/CreateAwsSagemakerNotebookStep.java

```diff
@@ -106,7 +106,6 @@ public StepResult undoStep(FlightContext flightContext) throws InterruptedExcept
             awsCloudContextService.discoverEnvironment());
 
     try {
-
       AwsUtils.stopSageMakerNotebook(credentialsProvider, resource);
```

Comment:
This just occurred to me (though it's from last PR) ... is `stopSageMakerNotebook` idempotent?  I.e. if we crash between line 109 and 112, will retries fail because it's already STOPPING/STOPPED?

Same question for delete on line 112, if the flight gets restarted after delete returns but before DELETING status?